### PR TITLE
feat(DMS): add datasource rabbitmq instances

### PR DESCRIPTION
**Pull request by 346010762**
**What this PR does / why we need it**:
   add datasource to query RabbitMQ instances by Terraform
**Special notes for your reviewer**:

**Release note**:

```release-note
   add datasource to query RabbitMQ instances by Terraform
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dms/ TESTARGS='-run TestAccDatasourceDmsRabbitMQInstances_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccDatasourceDmsRabbitMQInstances_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceDmsRabbitMQInstances_basic
=== PAUSE TestAccDatasourceDmsRabbitMQInstances_basic
=== CONT  TestAccDatasourceDmsRabbitMQInstances_basic
--- PASS: TestAccDatasourceDmsRabbitMQInstances_basic (757.32s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       757.367s
```

### CODE REVIEW

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rabbitmq_instances.go [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+	}
+
+	getRabbitmqInstancesPath := getRabbitmqInstancesClient.Endpoint + getRabbitmqInstancesHttpUrl
+	getRabbitmqInstancesPath = strings.ReplaceAll(getRabbitmqInstancesPath, "{project_id}", getRabbitmqInstancesClient.ProjectID)
+
+	getRabbitmqInstancesQueryParams := buildGetRabbitmqInstancesQueryParams(d)
+	getRabbitmqInstancesPath += getRabbitmqInstancesQueryParams
+
+	getRabbitmqInstancesOpt := golangsdk.RequestOpts{KeepResponseBody: true}
+	getRabbitmqInstancesResp, err := getRabbitmqInstancesClient.Request("GET", getRabbitmqInstancesPath, &getRabbitmqInstancesOpt)
```
Comment (houpeng80) [inline huaweicloud/services/dms/data_source_huaweicloud_dms_rabbitmq_instances.go]: only the first page is quired

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rabbitmq_instances.go [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+			"vpc_id":                     utils.PathSearch("vpc_id", v, nil),
+			"vpc_name":                   utils.PathSearch("vpc_name", v, nil),
+		})
+	}
+	return rst
+}
+
+func buildGetRabbitmqInstancesQueryParams(d *schema.ResourceData) string {
+	res := ""
+	res = fmt.Sprintf("%s&engine=%v", res, "rabbitmq")
```
Comment (houpeng80) [inline huaweicloud/services/dms/data_source_huaweicloud_dms_rabbitmq_instances.go]:
```suggestion
	res := fmt.Sprintf("?engine=%v", "rabbitmq")
```

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rabbitmq_instances.go [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+		res = fmt.Sprintf("%s&status=%v", res, v)
+	}
+
+	if v, ok := d.GetOk("enterprise_project_id"); ok {
+		res = fmt.Sprintf("%s&enterprise_project_id=%v", res, v)
+	}
+
+	if res != "" {
+		res = "?" + res[1:]
+	}
```
Comment (houpeng80) [inline huaweicloud/services/dms/data_source_huaweicloud_dms_rabbitmq_instances.go]: delete it

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rabbitmq_instances.go (hunk 1 above)
Comment (346010762) [reply huaweicloud/services/dms/data_source_huaweicloud_dms_rabbitmq_instances.go]: done

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rabbitmq_instances.go (hunk 2 above)
Comment (346010762) [reply huaweicloud/services/dms/data_source_huaweicloud_dms_rabbitmq_instances.go]: done

File: huaweicloud/services/dms/data_source_huaweicloud_dms_rabbitmq_instances.go (hunk 3 above)
Comment (346010762) [reply huaweicloud/services/dms/data_source_huaweicloud_dms_rabbitmq_instances.go]: done

**Pull request merged**